File: app/src/main/java/rbsoftware/friendstagram/model/Post.java
package rbsoftware.friendstagram.model;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

/**
 * Created by silver_android on 06/10/16.
 */

public class Post {
    private int id;
    @SerializedName("image_url")
    private String imageURL;
    @SerializedName("description")
    private String description;
    @SerializedName("tags")
    private ArrayList<String> hashTags;
    @SerializedName("user")
    private User user;

    public Post(String imageURL, String description, ArrayList<String> tags) {
        this.imageURL = imageURL;
        this.description = description;
        hashTags = new ArrayList<>(tags);
    }

    /**
     * Temporary constructor until server is up
     * @param imageURL
     * @param description
     */
    public Post(String imageURL, String description, User user) {
        this.imageURL = imageURL;
        this.description = description;
        this.user = user;
    }

    public int getID() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public User getUser() {
        return user;
    }

    public String getImageURL() {
        return imageURL;
    }

    public void setImageURL(String imageURL) {
        this.imageURL = imageURL;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof Post) {
            Post post = (Post) obj;
            return post.id == id && post.imageURL.equals(imageURL) && post.description.equals(description) && post.user.equals(user) && post.hashTags.equals(hashTags);
        }
        return false;
    }
}

File: app/src/main/java/rbsoftware/friendstagram/model/Post.kt
package rbsoftware.friendstagram.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

/**
 * Created by Rushil on 8/18/2017.
 */
data class Post(@SerializedName("url") val imageURL: String, val description: String) : Parcelable {
    var id = 0
    @SerializedName("tags")
    var hashtags: List<String> = listOf()
    lateinit var user: User

    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString()) {
        user = parcel.readParcelable(User::class.java.classLoader)
        id = parcel.readInt()
        parcel.readStringList(hashtags)
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(imageURL)
        parcel.writeString(description)
        parcel.writeParcelable(user, flags)
        parcel.writeInt(id)
        parcel.writeStringList(hashtags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Post> {
        override fun createFromParcel(parcel: Parcel): Post {
            return Post(parcel)
        }

        override fun newArray(size: Int): Array<Post?> {
            return arrayOfNulls(size)
        }
    }
}